ecodse_funtime_alpha/train.py

```diff
@@ -20,13 +20,14 @@ def train_loop(dataset, model, optimizer):
 
 
 def fit_loop(dataset, model, optimizer, nepoch, batchsize):
+    nstep = tf.data.experimental.cardinality(dataset).numpy() // batchsize
     dataset = dataset.shuffle(buffer_size=12)
```

Comment:
The buffer size should be much bigger. maybe something like `batchsize * 10`

Reply:
we probably want https://www.tensorflow.org/api_docs/python/tf/data/Dataset#prefetch

Reply:
replaced with batchsize * 10